File: Difficulty: Medium/Kadane's Algorithm/kadanes-algorithm.cs
class Solution {
    public int maxSubarraySum(int[] arr) {
        // code here
        int res = arr[0];
        int max= arr[0];
        for(int i=1;i<arr.Length;i++){
            max = Math.Max(max+arr[i], arr[i]);
            res = Math.Max(max,res);
        }
        return res;
    }
}

File: Difficulty: Medium/Kadane's Algorithm/kadanes-algorithm.java
class Solution {
    int maxSubarraySum(int[] arr) {
        // Code here
        int res = arr[0];
        int max = arr[0];
        for( int i = 1; i <arr.length;i++){
            max = Math.max(max + arr[i],arr[i]);
            res = Math.max(max, res);
        }
        
        return res;
        
    }
}
